toucan_connectors/odbc/odbc_connector.py

```diff
@@ -34,11 +35,18 @@ def _retrieve_data(self, datasource: ODBCDataSource) -> pd.DataFrame:
         connection = pyodbc.connect(self.connection_string, **self.get_connection_params())
 
         query_params = datasource.parameters or {}
-        # Apparently ODBC only supports qmark paramstyle
-        # (https://www.python.org/dev/peps/pep-0249/#paramstyle)
-        # so it expects a list of values, not a dict:
-        query_params = list(query_params.values())
-        df = pd.read_sql(datasource.query, con=connection, params=query_params)
+
+        # <FIXME: UNSAFE>
+        query_params = {k: f'"{v}"' for k, v in query_params.items()}  # add enclosing quotes
```

Comment:
Wouldn't it be safer to transform the query string itself and insert question marks so that we could still let `pyodbc` escape and substitute the values?